src/tests/Microsoft.PowerFx.Interpreter.Tests/CustomFunctions.cs

```diff
@@ -44,6 +44,95 @@ public static StringValue Execute(NumberValue x, BooleanValue b)
             }
         }
 
+        [Fact]
+        public async void CustomFunction_CallBack()
+        {
+            var config = new PowerFxConfig(null);
+            config.AddFunction(new TestCallbackFunction());
+            var engine = new RecalcEngine(config);
+
+            // Shows up in enuemeration
+            var func = engine.GetAllFunctionNames().First(name => name == "TestCallback");
+            Assert.NotNull(func);
+
+            var result = engine.Eval("TestCallback(1=2)");
```

Comment:
>1=2 [](http://example.com/codeflow?start=51&length=3)

We should find some way to demonstrate that this is actually a callback. 
ie:

// inc counter, return true on 3rd invoke
var c = new CounterFunction(); 

engine.AddFunction(c); 
engine.Eval("Wait(CounterFunction() = 3)");

Reply:
You could generalize that... 

int c = 0;
var func = new HelperFunction() {
  OnInvoke = () => { 
// could be any lambda, useful for reusing with tests. 
   c++;  return FormulaValue.New(c ==3);
 }
};

engine.AddFunction(func); 
engine.Eval("Wait(Helper() = 3)");

Reply:
also good to show a lambda binding against a variable (and not just be constant)